public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DatePicker from '../FormFields/DatePicker';
+import moment from 'moment';
+import Icon from '../Icon';
+
+export default class ScheduledLaunch extends React.Component {
+  static propTypes = {
+    video: PropTypes.object.isRequired,
+    saveVideo: PropTypes.func.isRequired,
+    videoEditOpen: PropTypes.bool.isRequired
+  };
+
+  state = {
+    selectedDate: null,
+    showDatePicker: false,
+    showScheduleButton: true
+  };
+
+  saveScheduledLaunch = () => {
+    const video = this.props.video;
+    this.props.saveVideo(Object.assign({}, video, {
+        contentChangeDetails: Object.assign({}, video.contentChangeDetails, {
+          scheduledLaunch: Object.assign({}, video.contentChangeDetails.scheduledLaunch, {
+            date: this.state.selectedDate
+          })
+        })
+      })
+    );
+    this.setState({ showDatePicker: false });
+  }
+
+  removeScheduledLaunch = () => {
+    const video = this.props.video;
+      this.props.saveVideo(Object.assign({}, video, {
+        contentChangeDetails: Object.assign({}, video.contentChangeDetails, {
+          scheduledLaunch: null
+        })
+      })
+    );
+    this.setState({ showDatePicker: false });
+  }
+
+
+  render() {
+    const { video, video: { contentChangeDetails }, videoEditOpen } = this.props;
+    const { selectedDate } = this.state;
+    const showDatePicker = this.state.showDatePicker && !videoEditOpen;
+    const isFutureDate = selectedDate && moment(selectedDate).isAfter(moment());
+    const scheduledLaunch = contentChangeDetails && contentChangeDetails.scheduledLaunch && contentChangeDetails.scheduledLaunch.date;
+    return (
+      <div className="flex-container topbar__scheduler">
+        {scheduledLaunch && !showDatePicker && <span className="topbar__launch-label">Scheduled: {moment(scheduledLaunch).format('Do MMM YYYY HH:mm')}</span> }
+        {
+          showDatePicker &&
+          <DatePicker
+            editable={true}
+            onUpdateField={(date) => this.setState({ selectedDate: date })}
```

Comment:
Could we not use the `updateVideo` function here that saves the date to the redux store but not the database?

Reply:
I decided to use the component's state to avoid incurring in the situation in which we have the value in the database out of sync with the value in the redux store. In this component's logic we are checking against the existence of that value in the database in order to show the scheduler vs the existing saved date. If the value in the store changed every time the value is updated but not saved, these checks would not work. To change this would require us to change the structure of the redux store, so I think keeping like this is easier.

Reply:
👍